test/e2e/set/scyllacluster/scyllacluster_ipv6.go

```diff
@@ -0,0 +1,338 @@
+					},
+					Clients: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypePodIP,
+					},
+				},
+			},
+			validateIPFamily: validateDualStackIPv4Primary,
+			validateScyllaAddrs: func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod) {
+				validatePodIPBroadcastAddresses(ctx, configClient, pod, corev1.IPv4Protocol)
+			},
+		}),
+		g.Entry("dual-stack (IPv6 primary) with PodIP broadcast", &ipv6TestEntry{
+			ipFamilyPolicy: corev1.IPFamilyPolicyPreferDualStack,
+			ipFamilies:     []corev1.IPFamily{corev1.IPv6Protocol, corev1.IPv4Protocol},
+			exposeOptions: &scyllav1.ExposeOptions{
+				BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+					Nodes: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypePodIP,
+					},
+					Clients: scyllav1.BroadcastOptions{
+						Type: scyllav1.BroadcastAddressTypePodIP,
+					},
+				},
+			},
+			validateIPFamily: validateDualStackIPv6Primary,
+			validateScyllaAddrs: func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod) {
+				validatePodIPBroadcastAddresses(ctx, configClient, pod, corev1.IPv6Protocol)
+			},
+		}),
+	)
+})
+
+func validateIPv6SingleStack(ctx context.Context, f *framework.Framework, sc *scyllav1.ScyllaCluster) {
+	pods, err := f.KubeClient().CoreV1().Pods(sc.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(naming.ClusterLabelsForScyllaCluster(sc)).String(),
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(pods.Items).ToNot(o.BeEmpty())
+
+	for _, pod := range pods.Items {
+		framework.By("Verifying pod %q has IPv6 address", pod.Name)
+
+		hasIPv6 := false
+		var ipv6Address string
+		for _, podIP := range pod.Status.PodIPs {
+			if ip, err := helpers.ParseIP(podIP.IP); err == nil && helpers.IsIPv6(ip) {
+				hasIPv6 = true
+				ipv6Address = podIP.IP
+				break
+			}
+		}
+		o.Expect(hasIPv6).To(o.BeTrue(), "Pod should have at least one IPv6 address")
+		framework.Infof("Pod %q has IPv6 address: %s", pod.Name, ipv6Address)
+	}
+
+	services, err := f.KubeClient().CoreV1().Services(sc.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(naming.ClusterLabelsForScyllaCluster(sc)).String(),
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	for _, svc := range services.Items {
+		framework.By("Verifying service %q IPv6 configuration", svc.Name)
+
+		hasIPv6Family := false
+		for _, family := range svc.Spec.IPFamilies {
+			if family == corev1.IPv6Protocol {
+				hasIPv6Family = true
+				break
+			}
+		}
+		o.Expect(hasIPv6Family).To(o.BeTrue(), "Service should have IPv6 in IPFamilies")
+
+		if svc.Spec.ClusterIP != corev1.ClusterIPNone {
+			clusterIP, err := helpers.ParseIP(svc.Spec.ClusterIP)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(helpers.IsIPv6(clusterIP)).To(o.BeTrue(), "Service ClusterIP should be IPv6: %s", svc.Spec.ClusterIP)
+		}
```

Comment:
I believe that we should either expect it to be populated or not. Please consider defining it on a test spec level, not deriving from the actual service we verify. Perhaps an `expectClusterIP` boolean passed to `validateIPv6SingleStack` will suffice.